gitlab/v4/objects.py

```diff
@@ -4712,6 +4712,74 @@ def import_project(
             "/projects/import", post_data=data, files=files, **kwargs
         )
 
+    def import_github(
+        self,
+        personal_access_token,
+        repo_id,
+        target_namespace,
+        new_name=None,
+        timeout_override=60.0,
+        **kwargs
+    ):
+        """Import a project from Github to Gitlab (schedule the import)
+
+        This method will return when an import operation has been safely queued,
+        or an error has occurred. After triggering an import, check the
+        `import_status` of the newly created project to detect when the import
+        operation has completed.
+
+        NOTE: this request may take longer than most other API requests.
+        So this method will override the session timeout with <timeout_override>,
+        which defaults to 60 seconds.
+
+        If repo_id is specified as an integer, it will be used directly.
+        If repo_id is a string, we will attempt to look it up using PyGithub.
+        The preferred string format is <org>/<repository>. However the python
+        package `PyGithub` must be installed to perform this lookup.
+
+        Args:
+            personal_access_token (str): GitHub personal access token
+            repo_id (int): Github repository ID
+            target_namespace (str): Namespace to import repo into
+            new_name (str): New repo name (Optional)
+            timeout_override (int or float): Timeout to use for this request
+            **kwargs: Extra options to send to the server (e.g. sudo)
+
+        Raises:
+            GitlabAuthenticationError: If authentication is not correct
+            GitlabListError: If the server failed to perform the request
+
+        Returns:
+            dict: A representation of the import status.
+
+        Example:
+        ```
+            gl = gitlab.Gitlab_from_config()
+            print "Triggering import"
+            result = gl.projects.import_github(ACCESS_TOKEN,
+                                               123456,
+                                               "my-group/my-subgroup")
+            project = gl.projects.get(ret['id'])
+            print "Waiting for import to complete"
+            while project.import_status == u'started':
+                time.sleep(1.0)
+                project = gl.projects.get(project.id)
+            print "Github import complete"
+        ```
+        """
+        data = {
+            "personal_access_token": personal_access_token,
+            "repo_id": repo_id,
+            "target_namespace": target_namespace,
+        }
+        if new_name:
+            data["new_name"] = new_name
+        prev_timeout = self.gitlab.timeout
+        self.gitlab.timeout = timeout_override
```

Comment:
This seems hacky. Can't you just provide timeout in the post method? (haven't looked into the code too much)

```python
kwargs["timeout"] = 60
result = self.gitlab.http_post("/import/github", post_data=data, **kwargs)
```

Reply:
I'm afraid not. It looks like on each invocation of `http_post` (or `http_request`) the timeout is freshly retrieved from `_get_session_opts()` and therefore pulled from the base gitlab object:

https://github.com/python-gitlab/python-gitlab/blob/master/gitlab/__init__.py#L490-L493

If I were to amend this behavior to also accept a timeout from `kwargs`, would this MR be an appropriate enough place to do it?

Reply:
ah yes. Feel free to create a new commit with this.